backend/models/task.model.ts

```diff
@@ -0,0 +1,48 @@
+import mongoose, { Schema, Document } from "mongoose";
+
+import { TaskStatus } from "../types";
+
+export interface Task extends Document {
+  id: string;
+  title: string;
+  description: string;
+  requiresApproval: boolean;
+  status: TaskStatus;
+  document: string;
+  dueDate: Date;
+  expiry: Date;
+}
+
+const TaskSchema: Schema = new Schema({
+  title: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+    required: true,
+  },
+  requiresApproval: {
+    type: Boolean,
+    required: true,
+  },
+  status: {
+    type: String,
```

Comment:
~This can have type `TaskStatus`, imported from `backend/types.ts`~ Oh wait nevermind that's not right, ignore this 😅